
See upstream dev@subversion thread.  Fixes intermittent failure of
javahl tests, particularly reproducible on aarch64.

Message-ID: <11de5f5c-5059-b973-95a1-385e7913a63a@syntevo.com>
From: Alexandr Miloslavskiy

--- subversion-1.14.1/subversion/bindings/javahl/tests/org/apache/subversion/javahl/BasicTests.java.fixjavatests
+++ subversion-1.14.1/subversion/bindings/javahl/tests/org/apache/subversion/javahl/BasicTests.java
@@ -4676,7 +4676,19 @@
             // RuntimeException("Test exception") is expected here
         }
 
-        tunnelAgent.joinAndTest();
+        // In this test, there is a race condition that sometimes results in
+        // IOException when 'WAIT_TUNNEL' tries to read from a pipe that
+        // already has its read end closed. This is not an error, but
+        // it's hard to distinguish this case from other IOException which
+        // indicate a problem. To reproduce, simply wrap this test's body in
+        // a loop. The workaround is to ignore any detected IOException.
+        //
+        // tunnelAgent.joinAndTest();
+        try {
+            tunnelAgent.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace ();
+        }
     }
 
     /**
